src/reducers/scrollStateHelper.js

```diff
@@ -252,8 +252,10 @@ export function updateRowCount(state, { rowsCount }) {
   state.scrollContentHeight = rowsCount * state.rowHeight;
   state.rowOffsets = PrefixIntervalTree.uniform(rowsCount, state.rowHeight);
   state.storedHeights = new Array(rowsCount);
+  state.rows = new Array(rowsCount);
   for (var i = 0; i < rowsCount; i++) {
     state.storedHeights[i] = state.rowHeight;
+    state.rows[i] = i;
```

Comment:
Yeah this line was required for it to work. If I call updateVisibleRows instead, we still need to update the underlying row data somehow, because it's changed and the way things are it's trying to iterate over rows that aren't supposed to be there. This screen should help explain the problem:
![selection_003](https://user-images.githubusercontent.com/294747/26876240-7908baa2-4b42-11e7-9f66-d6ba09ca114e.png)
rowHeights is up to date, rows is not